lego/apps/users/views/users.py

```diff
@@ -173,3 +175,33 @@ def change_grade(self, request, *args, **kwargs):
                 newGrade.add_user(user)
 
         return Response(MeSerializer(user).data)
+
+    @action(
+        detail=True,
+        methods=["POST"],
+        permission_classes=[IsAuthenticated],
+        serializer_class=PhotoConsentSerializer,
+    )
+    def update_photo_consent(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        user = self.get_object()
+
+        with transaction.atomic():
+            photo_consent = user.photo_consents.get(
+                year=serializer.validated_data["year"],
+                semester=serializer.validated_data["semester"],
+                domain=serializer.validated_data["domain"],
+            )
+
+            if not request.user.has_perm(EDIT, user):
+                raise ValueError("Cannot update other user's consent")
```

Comment:
Any reason for fetching the photo consent before doing the permission check?

Reply:
Not really. Fixed